File: android/src/main/java/com/valentingrigorean/arcgis_maps_flutter/map/SelectionPropertiesHandler.java
package com.valentingrigorean.arcgis_maps_flutter.map;

import android.graphics.Color;

import com.esri.arcgisruntime.mapping.SelectionProperties;
import com.esri.arcgisruntime.mapping.view.Graphic;

public class SelectionPropertiesHandler {
    private final SelectionProperties selectionProperties;
    private final int defaultSelectedColor;

    public SelectionPropertiesHandler(SelectionProperties selectionProperties) {
        this.selectionProperties = selectionProperties;
        this.defaultSelectedColor = selectionProperties.getColor();
    }

    public void setSelected(Graphic graphic, Color selectedColor) {
        if (selectedColor != null) {
            selectionProperties.setColor(selectedColor.toArgb());
        } else {
            reset();
        }
        graphic.setSelected(true);
    }

    public void clearSelection(Graphic graphic) {
        graphic.setSelected(false);
        reset();
    }

    public void reset() {
        if (selectionProperties.getColor() == defaultSelectedColor)
            return;
        selectionProperties.setColor(defaultSelectedColor);
    }
}

File: android/src/main/java/com/valentingrigorean/arcgis_maps_flutter/map/SelectionPropertiesHandler.kt
package com.valentingrigorean.arcgis_maps_flutter.map

import android.graphics.Color
import com.arcgismaps.mapping.view.Graphic
import com.arcgismaps.mapping.view.SelectionProperties

class SelectionPropertiesHandler(private val selectionProperties: SelectionProperties) {
    private val defaultSelectedColor: com.arcgismaps.Color = selectionProperties.color

    fun setSelected(graphic: Graphic, selectedColor: Color?) {
        if (selectedColor != null) {
            selectionProperties.color = com.arcgismaps.Color(selectedColor.toArgb())
        } else {
            reset()
        }
        graphic.isSelected = true
    }

    fun clearSelection(graphic: Graphic) {
        graphic.isSelected = false
        reset()
    }

    fun reset() {
        if (selectionProperties.color == defaultSelectedColor) return
        selectionProperties.color = defaultSelectedColor
    }
}